streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -1500,68 +1507,57 @@ private void validateRackAwarenessConfiguration() {
         });
     }
 
+    private void overwritePropertyMap(final Map<String, Object> props, final String key, final Object unmodifiableDefaultValue, final String config) {
+        final String overwritePropertyLogMessage = "Unexpected %s config `%s` found. User setting (%s) will be ignored and the Kafka Streams default setting (%s) will be used";
+        
+        if (props.containsKey(key) && (!Objects.equals(props.get(key), unmodifiableDefaultValue))) {
+            log.warn(String.format(overwritePropertyLogMessage, config, key, props.get(key), unmodifiableDefaultValue));
```

Comment:
For `transaction.id` we would log `[...] Kafka Streams default setting (null) will be used"`. But this not accurate. In the end, we use
- eos_v1: `<application.id>-<task_id>`
- eos_v2: `<application.id>-<processId>-<threadIdx>`

I think it's worth to put some if-else in here to have special handling for `transactional.id` and not print `null` but one of both expressions from above depending which version of eos is used.

Reply:
Got it - I have made this change in the manner that we now have two function definitions for `overwritePropertyMap`
1.  `overwritePropertyMap(Map props, String configName, Object unmodifiableValue, Object unmodifiableLogValue, String clientType)`
2. `overwritePropertyMap(Map props, String configName, Object unmodifiableValue, String clientType)`

Since the scope of `overwritePropertyMap` is to both log the warning and override the props map it was odd to populate the props with the suggested expressions. 

Hence in the scenario of `transaction.id` I have opted to set the value still as `null` and log the value as the expressions.

Can you please have a look at the implementation and suggest any improvements if possible?